### DISCUSSION TITLE
Tasmota with scripting --- Unable to change device name

### DISCUSSION THREAD

**Discussion by doceave** [arendst/Tasmota#17702] [General]
After having successfully flashed a device with Tasmota with scripting enabled I am unable to change the device name... Or rather it defaults back to "Tasmota" after reboot.

![Screenshot 2023-01-14 192305](https://user-images.githubusercontent.com/323727/212486714-40aab056-e698-40b0-bf95-39f284f85f84.jpg)

Is this a known problem or something I am doing incorrectly? It does not look like the script is enforcing the naming: https://gist.github.com/thxthx0/c22555f6ec25c3367264729a0ff74dce

**Comment by barbudor** [comment]
Indeed, you can change it using the command `DeviceName` in the console but something goes wrong if you try to change through the web page
If you disable the script, then it works

something weird ...

**Reply by barbudor** [reply]
In my case, when I click on Save (while script is active), the WebServer doesn't seem to respond properly. I end up with a page saying "The site can't be reached, the connection was reset"

It looks like while the script is running, the WebServer is partially disturbed

No idea why

Buty I as told you can still change the DeviceName
- Using the web console command
- By the web UI but wit the script stopped
